src/extension.ts

```diff
@@ -6,51 +6,58 @@ import {
   IndentStyle,
   CommaPosition,
   LogicalOperatorNewline,
+  FormatOptions,
 } from 'sql-formatter';
 
 const getConfigs = (
-  settings: vscode.WorkspaceConfiguration,
-  formattingOptions: vscode.FormattingOptions | { tabSize: number; insertSpaces: boolean },
+  extensionSettings: vscode.WorkspaceConfiguration,
+  formattingOptions: Partial<vscode.FormattingOptions>,
   language: SqlLanguage
-) => {
-  const ignoreTabSettings = settings.get<boolean>('ignoreTabSettings');
-  const { tabSize, insertSpaces } = ignoreTabSettings // override tab settings if ignoreTabSettings is true
-    ? {
-        tabSize: settings.get<number>('tabSizeOverride')!,
-        insertSpaces: settings.get<boolean>('insertSpacesOverride')!,
-      }
-    : formattingOptions;
-  const indent = insertSpaces ? ' '.repeat(tabSize) : '\t';
-
-  // build format configs from settings
-  const formatConfigs = {
+): Partial<FormatOptions> => {
+  return {
     language:
       language === 'sql' // override default SQL language mode if SQLFlavourOverride is set
-        ? settings.get<SqlLanguage>('SQLFlavourOverride') ?? 'sql'
+        ? extensionSettings.get<SqlLanguage>('SQLFlavourOverride') ?? 'sql'
         : language,
-    indent,
-    keywordCase: settings.get<KeywordCase>('keywordCase'),
-    indentStyle: settings.get<IndentStyle>('indentStyle'),
-    logicalOperatorNewline: settings.get<LogicalOperatorNewline>('logicalOperatorNewline'),
-    tabulateAlias: settings.get<boolean>('tabulateAlias'),
-    commaPosition: settings.get<CommaPosition>('commaPosition'),
-    expressionWidth: settings.get<number>('expressionWidth'),
-    linesBetweenQueries: settings.get<number>('linesBetweenQueries'),
-    denseOperators: settings.get<boolean>('denseOperators'),
-    newlineBeforeSemicolon: settings.get<boolean>('newlineBeforeSemicolon'),
+    ...getIndentationConfig(extensionSettings, formattingOptions),
+    keywordCase: extensionSettings.get<KeywordCase>('keywordCase'),
+    indentStyle: extensionSettings.get<IndentStyle>('indentStyle'),
+    logicalOperatorNewline: extensionSettings.get<LogicalOperatorNewline>('logicalOperatorNewline'),
+    tabulateAlias: extensionSettings.get<boolean>('tabulateAlias'),
+    commaPosition: extensionSettings.get<CommaPosition>('commaPosition'),
+    expressionWidth: extensionSettings.get<number>('expressionWidth'),
+    linesBetweenQueries: extensionSettings.get<number>('linesBetweenQueries'),
+    denseOperators: extensionSettings.get<boolean>('denseOperators'),
+    newlineBeforeSemicolon: extensionSettings.get<boolean>('newlineBeforeSemicolon'),
   };
+};
 
-  return formatConfigs;
+const getIndentationConfig = (
+  extensionSettings: vscode.WorkspaceConfiguration,
+  formattingOptions: Partial<vscode.FormattingOptions>
```

Comment:
can we not just use FormattingOptions instead of partial for here and above ?

Reply:
Indeed, Partial not needed here. Will change.